src/HDF5Msg.chpl

```diff
@@ -1905,6 +1998,474 @@ module HDF5Msg {
+                                    dtype = getDataType(bool);
+                                }
+                                when (DType.BigInt) {
+                                    var values = toSymEntry(vals_entry, bigint);
+                                    // create the group
+                                    validateGroup(file_id, f, "%s/%s".format(group, SEGMENTED_VALUE_NAME), overwrite); // stored as group - group uses the dataset name
+                                    //localize values and write dataset
+                                    writeSegmentedLocalDset(file_id, group, values, segments, true, bigint);
+                                    dtype = getDataType(uint);
+                                }
+                                when (DType.Strings){
+                                    var values = toSegStringSymEntry(vals_entry);
+                                    // create the group
+                                    validateGroup(file_id, f, "%s/%s".format(group, SEGMENTED_VALUE_NAME), overwrite);
+                                    //localize values and write dataset
+                                    writeNestedSegmentedLocalDset(file_id, group, values, segments, true, uint(8));
+                                    dtype = getDataType(uint(8));
+                                }
+                                otherwise {
+                                    throw getErrorWithContext(
+                                    msg="Unsupported SegArray DType %s".format(str2dtype(dt)),
+                                    lineNumber=getLineNumber(),
+                                    routineName=getRoutineName(), 
+                                    moduleName=getModuleName(),
+                                    errorClass="IllegalArgumentError");
+                                }
+                            }
+                            writeArkoudaMetaData(file_id, group, objType:string, dtype);
+                        }
+                        otherwise {
+                            throw getErrorWithContext(
+                            msg="Unsupported ObjType %s".format(ot: string),
+                            lineNumber=getLineNumber(),
+                            routineName=getRoutineName(), 
+                            moduleName=getModuleName(),
+                            errorClass="IllegalArgumentError");
+                        }
+                    }
+                }
+
+                // write dataframe meta data
+                writeDataFrameMetaData(file_id, group, objType: string, num_cols);
+                C_HDF5.H5Fclose(file_id);
+            } 
+            when MULTI_FILE {
+                var filenames = prepFiles(filename, mode, idx_gen);
+
+                // need to add the group to all files
+                coforall (loc, idx) in zip(Locales, filenames.domain) do on loc {
+                    const localeFilename = filenames[idx];
```

Comment:
why don't we just loop over `filenames` instead looping the domain and then indexing?

Reply:
This needs to be done using the domain for proper functionality of the `c_str()` calls.

Reply:
makes sense!